contract/contracts/Officer.sol

```diff
@@ -0,0 +1,113 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import "@openzeppelin/contracts/proxy/beacon/BeaconProxy.sol";
+import "@openzeppelin/contracts/proxy/beacon/UpgradeableBeacon.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+import '@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol';
+import '@openzeppelin/contracts-upgradeable/utils/PausableUpgradeable.sol';
+import '@openzeppelin/contracts-upgradeable/utils/ReentrancyGuardUpgradeable.sol';
+
+import "hardhat/console.sol";
+
+interface IBankAccount {
+    function initialize(address tipsAddress) external;
+}
+
+interface IVotingContract {
+    function initialize() external;
+}
+
+contract Officer is OwnableUpgradeable, ReentrancyGuardUpgradeable, PausableUpgradeable  {
+    
+    address[] founders;
+    address[] members;
+    address bankAccountContract;
+    address votingContract;
+
+    address public bankAccountBeacon;
+    address public employeeContractBeacon;
+    address public votingContractBeacon;
+
+    event TeamCreated( address[] founders, address[] members);
+    event ContractDeployed( string contractType, address contractAddress);
+
+  function initialize( address _bankAccountBeacon, address _votingContractBeacon) public initializer {
+        __Ownable_init(msg.sender);
+        __ReentrancyGuard_init();
+        __Pausable_init();
+        bankAccountBeacon = _bankAccountBeacon;
+        votingContractBeacon = _votingContractBeacon;
+    }
+
+   function createTeam(
+    address[] memory _founders, 
+    address[] memory _members
+    ) external {
+        require(_founders.length > 0, "Must have at least one founder");
+        bankAccountContract = address(0);
+        votingContract = address(0);
```

Comment:
```suggestion
```
we can remove it if we will call it in initialize, and I think this function should be private and can be call once by initialize